Tools/cases_generator/analyzer.py

```diff
@@ -0,0 +1,458 @@
+        self.properties.dump("    " + indent)
+
+    @property
+    def size(self) -> int:
+        return 1 + sum(uop.size for uop in self.uops)
+
+
+@dataclass
+class PseudoInstruction:
+    name: str
+    targets: list[Instruction]
+    flags: list[str]
+
+    def dump(self, indent: str) -> None:
+        print(indent, self.name, "->", " or ".join([t.name for t in self.targets]))
+
+
+@dataclass
+class Family:
+    name: str
+    size: str
+    members: list[Instruction]
+
+    def dump(self, indent: str) -> None:
+        print(indent, self.name, "= ", ", ".join([m.name for m in self.members]))
+
+
+@dataclass
+class Analysis:
+    instructions: dict[str, Instruction]
+    uops: dict[str, Uop]
+    families: dict[str, Family]
+    pseudos: dict[str, PseudoInstruction]
+
+
+def analysis_error(message: str, tkn: lexer.Token) -> SyntaxError:
+    # To do -- support file and line output
+    # Construct a SyntaxError instance from message and token
+    return lexer.make_syntax_error(message, "", tkn.line, tkn.column, "")
+
+
+def override_error(
+    name: str,
+    context: parser.Context | None,
+    prev_context: parser.Context | None,
+    token: lexer.Token,
+) -> SyntaxError:
+    return analysis_error(
+        f"Duplicate definition of '{name}' @ {context} "
+        f"previous definition @ {prev_context}",
+        token,
+    )
+
+
+def convert_stack_item(item: parser.StackEffect) -> StackItem:
+    return StackItem(item.name, item.type, item.cond, (item.size or "1"))
+
+
+def analyze_stack(op: parser.InstDef) -> StackEffect:
+    inputs: list[StackItem] = [
+        convert_stack_item(i) for i in op.inputs if isinstance(i, parser.StackEffect)
+    ]
+    outputs: list[StackItem] = [convert_stack_item(i) for i in op.outputs]
+    for input, output in zip(inputs, outputs):
+        if input.name == output.name:
+            input.peek = output.peek = True
+    return StackEffect(inputs, outputs)
+
+
+def analyze_caches(op: parser.InstDef) -> list[CacheEntry]:
+    caches: list[parser.CacheEffect] = [
+        i for i in op.inputs if isinstance(i, parser.CacheEffect)
+    ]
+    return [CacheEntry(i.name, int(i.size)) for i in caches]
+
+
+def variable_used(node: parser.InstDef, name: str) -> bool:
+    """Determine whether a variable with a given name is used in a node."""
+    return any(
+        token.kind == "IDENTIFIER" and token.text == name for token in node.tokens
+    )
+
+
+def is_infallible(op: parser.InstDef) -> bool:
+    return not (
+        variable_used(op, "ERROR_IF")
+        or variable_used(op, "error")
+        or variable_used(op, "pop_1_error")
+        or variable_used(op, "exception_unwind")
+        or variable_used(op, "resume_with_error")
+    )
+
+
+from flags import makes_escaping_api_call
+
+EXITS = set(
+    [
+        "DISPATCH",
+        "GO_TO_INSTRUCTION",
+        "Py_UNREACHABLE",
+        "DISPATCH_INLINED",
+        "DISPATCH_GOTO",
+    ]
+)
```

Comment:
Just use a literal set?

```suggestion
EXITS = {
    "DISPATCH",
    "GO_TO_INSTRUCTION",
    "Py_UNREACHABLE",
    "DISPATCH_INLINED",
    "DISPATCH_GOTO",
}
```